ginisdk/src/main/java/net/gini/android/SdkBuilder.java

```diff
@@ -181,23 +219,52 @@ public Gini build() {
      * Helper method to create (and store) the RequestQueue which is used for both the requests to the Gini API and the
      * Gini User Center API.
      *
-     * @return                      The RequestQueue instance.
+     * @return The RequestQueue instance.
      */
     private synchronized RequestQueue getRequestQueue() {
         if (mRequestQueue == null) {
-            RequestQueueBuilder requestQueueBuilder = new RequestQueueBuilder(mContext);
             if (mCache != null) {
-                requestQueueBuilder.setCache(mCache);
+                mRequestQueue = new RequestQueue(mCache, new BasicNetwork(getHttpStack()));
+            } else {
+                mRequestQueue = Volley.newRequestQueue(mContext, getHttpStack());
             }
-            mRequestQueue = requestQueueBuilder.build();
         }
         return mRequestQueue;
     }
 
+    /**
+     * Helper method to create a HttpStack which is used performs Http request. Optionally it can
+     * take care of certificate pinning functionality.
+     *
+     * @return HttpStack instance.
+     */
+
+    private synchronized HttpStack getHttpStack() {
+        HttpStack stack;
+
+        if (mCertificatePaths != null && mCertificatePaths.length > 0) {
+            try {
+                TrustManager tm[] = {new PubKeyManager(getLocalCertificatesFromAssets(mCertificatePaths))};
+
+                SSLContext context = SSLContext.getInstance("TLS");
+                context.init(null, tm, null);
+                SSLSocketFactory pinnedSSLSocketFactory = context.getSocketFactory();
+                stack = new HurlStack(null, pinnedSSLSocketFactory);
+
+            } catch (IOException | CertificateException | NoSuchAlgorithmException | KeyManagementException e) {
+                stack = new HurlStack();
```

Comment:
Network errors are handle throw Bolts tasks (`task.getError()`). This kind of errors should be handle by a developer, so do you think that it is worth to change the `build` method to make them catch it? Because it would imply changes for the clients.
That is why I just decided to print the error in the logs. What do you think?